WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -333,6 +352,26 @@ public bool IsUsed(HdPubKey hdPubKey)
 		}
 	}
 
+	/// <summary>Gets transaction amount representing change in wallet balance for the wallet the transaction belongs to.</summary>
+	/// <returns>The same value as <see cref="TransactionSummary.Amount"/>.</returns>
+	public bool TryGetTxAmount(uint256 txid, [NotNullWhen(true)] out Money? amount)
+	{
+		lock (Lock)
+		{
+			return TransactionAmountsByTxid.TryGetValue(txid, out amount);
+		}
+	}
+
+	/// <summary>Gets total balance as a sum of unspent coins.</summary>
+	public Money GetTotalBalance()
+	{
+		lock (Lock)
+		{
+			// Amount can be hold as a variable that is updated every time to avoid summing it.
```

Comment:
I believe the best way for this is to use a `TotalBalanceSnapshot`, and  update in `UpdateSnapshotsNoLock`